source/nijigenerate/package.d

```diff
@@ -129,7 +129,7 @@ void incClearImguiData() {
     Returns the current project path
 */
 string incProjectPath() {
-    return currProjectPath;
+    return currProjectPath.dup;
```

Comment:
I couldn't see entire part of the code now, but it seems duplicating every time in incProjectPath might be heavy operation if it is used in update loop of UI. if that happens, string duplication should be called at client side and only if it should be duplicated.

Reply:
It seems that this function is only called during file access, so  encapsulates and protects the internal resources of `incProjectPath`.

Additionally, the issue appears to be caused by `incShowSaveDialog()` and `incShowOpenDialog()`, as they may be called in other places as well. I will apply a patch there too. that only called once when the user needs.

Reply:
I found incProjectPath is called in every update loop in incCheckAutosave.
So dup should not be called inside incProjectPath.

I still don't understand why this dup is needed yet.

Reply:
I think the root cause is that after users save new files and export png or videos, they share the string returned by `incShowSaveDialog()`

leading to the two issues mentioned above. 
The crash on macOS differ from those on Windows, windows is access denied, macos indicates that it is a directory.

Reply:
This will cause the user to overwrite the incProjectPath after exporting the png.

Reply:
Maybe the d language cast is using the c_string memory address, I'm not sure

Reply:
This bug can also be reproduced like this

1. open inx
2. save as new inx
3. export png or video
4. Ctrl+s

You can see that the saved path is incorrect, but it will not crash.

Reply:
incCheckAutosave seems to be called per frame, that may need to be modified